lib/gemstash/gem_pusher.rb

```diff
@@ -53,6 +53,13 @@ def check_auth
     end
 
     def store_gem
+      resource_exist = storage.resource(full_name).exist?
+      resource_is_indexed = storage.resource(full_name).properties[:indexed] if resource_exist
+
+      if resource_exist && resource_is_indexed
```

Comment:
This guard is mirrored in [`save_to_database`](https://github.com/bundler/gemstash/blob/f14894610a1cbef21d6f595ba3b332e45baa5f32/lib/gemstash/gem_pusher.rb#L79). Perhaps it should be extracted to a private method?

Also, there is a [`YankedVersionError` guard](https://github.com/bundler/gemstash/blob/f14894610a1cbef21d6f595ba3b332e45baa5f32/lib/gemstash/gem_pusher.rb#L80) in `save_to_database`. Should that _also_ be here?

Reply:
WIll do a followup PR if applies, thanks